docs/README.md

```diff
@@ -225,6 +227,41 @@ const Home = () => {
 };
 ```
 
+## 🕒usePrevious
+
+Custom hook for retrieving the previous useState value
+
+### Usage
+
+```jsx
+function App() {
+  // normal usage of useState
+  const [visible, setVisible] = useState(false);
+  
+  // using the custom usePrevious hook to retrieve the value that was provided in the previous render
+  const prevVisibility = usePrevious(visible);
+  
+  // Display both current and previous visibility states
+  return (
+    <div>
+      <h1>Current visibility: {visibility}, previous visibility: {prevVisibility}</h1>
+      <button onClick={() => setVisible(!visible)}>Toggle Visibility</button>
+    </div>
+   );
+}
+```
+
+### Parameters
+
+1. `value` (_any_) : The current value (will be the previous value in the next render).
+
+### Return value
+
+`[value]`
```

Comment:
This is is not needed, as it will make the readers feel that this function returns an array.

```suggestion

```